packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  FoldedCollection,
+  FoldedResult,
+  FoldedResultList,
+  FoldedResultListState,
+  Result,
+  ResultList,
+  ResultListState,
+  ResultsPerPageState,
+  ResultTemplatesManager,
+  ResultTemplate,
+  buildResultTemplatesManager,
+  ResultListProps,
+  EcommerceDefaultFieldsToInclude,
+} from '@coveo/headless';
+import {identity} from 'lodash';
+import {Bindings} from '../../utils/initialization-utils';
+import {
+  ResultDisplayDensity,
+  ResultDisplayImageSize,
+  ResultDisplayLayout,
+} from '../atomic-result/atomic-result-display-options';
+import {TemplateContent} from '../result-templates/result-template-common';
+
+interface DisplayProps {
+  density: ResultDisplayDensity;
+  imageSize?: ResultDisplayImageSize;
+  image?: ResultDisplayImageSize;
+  display?: ResultDisplayLayout;
+}
+export interface ResultPlaceholderProps extends DisplayProps {
+  resultsPerPageState: ResultsPerPageState;
+}
+export interface ResultsProps extends DisplayProps {
+  resultListState: FoldedResultListState | ResultListState;
+  bindings: Bindings;
+  host: HTMLElement;
+  getContentOfResultTemplate(
+    result: Result | FoldedResult
+  ): HTMLElement | DocumentFragment;
+
+  classes: string;
+}
+
+export interface AtomicResultListBase extends Omit<ResultsProps, 'classes'> {
+  templateHasError: boolean;
+  listWrapperRef?: HTMLDivElement;
+  resultsPerPageState: ResultsPerPageState;
+
+  resultListCommon?: ResultListCommon;
+
+  resultList: FoldedResultList | ResultList;
+}
+
+export function getId(
```

Comment:
All the methods/variables that exported here should have a more specific name e.g. getAtomicResultListId.
If it's too generic, using the IDE you'll get completions that can be a bit unclear,

Reply:
I would also move as many of these as possible inside the class below, even if they have no parameters.